src/python/pants/backend/native/subsystems/native_toolchain.py

```diff
@@ -161,6 +163,7 @@ def select_llvm_c_toolchain(platform, native_toolchain):
   llvm_c_compiler_args = [
     '-x', 'c', '-std=c11',
     '-nobuiltininc',
+    '-nostdinc',
```

Comment:
You've added this in four places and it seems related to the failing unit tests:
+ TestNativeToolchain.test_hello_c_clang
+ TestNativeToolchain.test_hello_c_gcc
+ TestNativeToolchain.test_hello_cpp_clangpp
+ TestNativeToolchain.test_hello_cpp_gpp

As well as in the TestProjectsIntegrationTest.test_shard_44 integration test which shows:
```
...
                     E   	03:49:57 00:01   [native-compile]
                     E   	03:49:57 00:01     [native-third-party-fetch]
                     E   	                   Invalidated 2 targets.
                     E   	03:50:11 00:15     [c-for-ctypes]
                     E   	03:50:11 00:15     [cpp-for-ctypes]
                     E   	                   Invalidated 2 targets.
                     E   	                   selected compiler exe name: 'clang++'
                     E   	03:50:11 00:15       [cpp-compile]
                     E   	                     clang-6.0: warning: argument unused during compilation: '-nobuiltininc' [-Wunused-command-line-argument]
                     E   	                     /home/travis/build/pantsbuild/pants/testprojects/src/python/python_distribution/ctypes_with_extra_compiler_flags/some_more_math.cpp:2:10: fatal error: 'assert.h' file not found
                     E   	                     #include <assert.h>
                     E   	                              ^~~~~~~~~~
                     E   	                     1 error generated.
...
```

And CTypesIntegrationTest.test_ctypes_third_party_integration which shows:
```
...
                     E   	03:21:58 00:01   [native-compile]
                     E   	03:21:58 00:01     [native-third-party-fetch]
                     E   	                   Invalidated 2 targets.
                     E   	03:22:11 00:14     [c-for-ctypes]
                     E   	03:22:12 00:15     [cpp-for-ctypes]
                     E   	                   Invalidated 1 target.
                     E   	                   selected compiler exe name: 'clang++'
                     E   	03:22:12 00:15       [cpp-compile]
                     E   	                     clang-6.0: warning: argument unused during compilation: '-nobuiltininc' [-Wunused-command-line-argument]
                     E   	                     In file included from /home/travis/build/pantsbuild/pants/testprojects/src/python/python_distribution/ctypes_with_third_party/some_more_math_with_third_party.cpp:4:
                     E   	                     /home/travis/build/pantsbuild/pants/.pants.d/tmp/tmpQX3l3R.pants.d/native-compile/native-third-party-fetch/a33349f38f6e/testprojects.src.python.python_distribution.ctypes_with_third_party.rang/current/include/rang.hpp:15:10: fatal error: 'unistd.h' file not found
                     E   	                     #include <unistd.h>
                     E   	                              ^~~~~~~~~~
                     E   	                     1 error generated.
...
```